### PR TITLE
refactor(jsonrpc): Removed unnecessary `mut` requirement for the RPC client methods

### PR DESCRIPTION
**Pull request by frol**

### CODE REVIEW

File: chain/jsonrpc/tests/test_utils.rs [hunk 1]
```diff
@@ -56,3 +56,22 @@ pub fn start_all_with_validity_period(
...
     (view_client_addr, addr)
 }
+
+#[allow(unused_macros)] // Suppress Rustc warnings even though this macro is used.
+macro_rules! test_with_client {
+    ($validator_status:expr, $client:ident, $block:expr) => {
+        init_test_logger();
+
+        System::run(|| {
+            let (_view_client_addr, addr) = test_utils::start_all($validator_status == "validator");
```
Comment (bowenwang1996) [inline chain/jsonrpc/tests/test_utils.rs:66]: this feels a bit weird. Why not just use a boolean?

Comment (frol) [reply chain/jsonrpc/tests/test_utils.rs:66]: If I use boolean than the usage will be:

```rust
test_with_client!(false, client, ...);
```

This `false` is not self-descriptive and I have to look it up. I may introduce an enum, though.

Comment (bowenwang1996) [reply chain/jsonrpc/tests/test_utils.rs:66]: Right but I feel like your way is no better than using boolean. It was a bit confusing to me at first. I think definitely enum would be better if you want to be more explicit.

File: chain/jsonrpc/tests/rpc_query.rs [hunk 2]
```diff
@@ -14,30 +14,13 @@ use near_primitives::test_utils::init_test_logger;
...
-        })
-        .unwrap();
-    };
-}
-
 /// Retrieve blocks via json rpc
 #[test]
 fn test_block_by_id_height() {
-    test_with_client!(client, async move {
+    test_with_client!(test_utils::NodeType::NonValidator, client, async move {
```
Comment (bowenwang1996) [inline chain/jsonrpc/tests/rpc_query.rs:23]: Why not `use test_utils::NodeType`?

Comment (frol) [reply chain/jsonrpc/tests/rpc_query.rs:23]: I just wanted to make sure it is clear that this is an ad-hoc enum, not something that is a part of the core types.

**Pull request merged**